src/main/java/nextstep/ladders/domain/LadderDetail.java

```diff
@@ -0,0 +1,36 @@
+package nextstep.ladders.domain;
+
+import nextstep.ladders.domain.exceptions.DifferentCountException;
+
+import java.util.Arrays;
+
+public class LadderDetail {
+
+    private final Participants participants;
+    private final ExecutionResults executionResults;
+
+    public LadderDetail(final String participantsText, final String executionResultText) {
+        checkEqualsCount(participantsText, executionResultText);
+        this.participants = new Participants(participantsText);
+        this.executionResults = new ExecutionResults(executionResultText);
+    }
```

Comment:
```suggestion
    private LadderDetail(final Participants participantsText, final ExecutionResults executionResultText) {
        this.participants = participants;
        this.executionResults = executionResults;
    }
    
    public static LadderDetail of (final Participants participantsText, final ExecutionResults executionResultText) {
        return new LadderDetail(participantsText, executionResultText);
    }
    
    public static LadderDetail of (final String participantsText, final String executionResultText) {
        checkEqualsCount(participantsText, executionResultText);
        return of(new Participants(participantsText), new ExecutionResults(executionResultText));
    }
```
이렇게 `정적 생성 메소드 팩토리 패턴`을 이용해보면 어떨까요?